### PR TITLE
Feature/ftm task level

### PR DESCRIPTION
**Pull request by rishabh-arya95**
# Issue Link

Resolves https://github.com/LambdaTest/test-at-scale/issues/127

# Description

Create FTM mode in nuclues

Fixes # (issue)

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested using premerge webhook.
- [x] Tested using postmerge webhook.
- [x] Tested auto quarantine feature.
- [x] Tested by deliberately failing discovery, execution and ftm tasks.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: config/nucleusmodel.go [hunk 1]
```diff
@@ -6,23 +6,26 @@ import "github.com/LambdaTest/test-at-scale/pkg/lumber"
...
+	Port            string
+	PayloadAddress  string `json:"payloadAddress" yaml:"payloadAddress"`
+	CollectStats    bool   `json:"collectStats"`
+	ConsecutiveRuns int    `json:"consecutiveRuns"`
+	LogFile         string
+	LogConfig       lumber.LoggingConfig
+	CoverageMode    bool   `json:"coverage" yaml:"coverageOnly"`
+	DiscoverMode    bool   `json:"discover" yaml:"discoverOnly"`
+	ExecuteMode     bool   `json:"execute" yaml:"executeOnly"`
+	FlakyMode       bool   `json:"flaky" yaml:"flakyOnly"`
```
Comment (saurabh-prakash) [inline config/nucleusmodel.go:19]: Let's remove `yaml` tags. I think, they aren't unused

File: pkg/core/lifecycle.go [hunk 2]
```diff
@@ -80,11 +80,17 @@ func (pl *Pipeline) Start(ctx context.Context) (err error) {
...
 	if pl.Cfg.DiscoverMode {
 		taskPayload.Type = DiscoveryTask
+	} else if pl.Cfg.FlakyMode {
+		mode = "flaky"
+		taskPayload.Type = FlakyTask
 	} else {
+		mode = "execution"
 		taskPayload.Type = ExecutionTask
 	}
+	pl.Logger.Infof("Running nucleus in %s mode", mode)
```
Comment (saurabh-prakash) [inline pkg/core/lifecycle.go:93]: instead of `mode` (a new variable), can we reuse `taskPayload.Type` for the info log?

**Review by saurabh-prakash — COMMENTED** [review]
minor comments. Else, lgtm

**Review by palashmarantas — DISMISSED** [review]

File: pkg/core/lifecycle.go (hunk 2 above)
Comment (rishabh-arya95) [reply pkg/core/lifecycle.go]: done

File: config/nucleusmodel.go (hunk 1 above)
Comment (rishabh-arya95) [reply config/nucleusmodel.go]: done

**Review by saurabh-prakash — APPROVED** [review]

**Review by palashmarantas — APPROVED** [review]

**Pull request merged**